services/kube-eleven/server/domain/utils/kube-eleven/kube_eleven.go

```diff
@@ -153,16 +155,67 @@ func (k *KubeEleven) generateTemplateData() templateData {
 
 	var potentialEndpointNode *spec.Node
 	data.Nodepools, potentialEndpointNode = k.getClusterNodes()
-
 	data.APIEndpoint = k.findAPIEndpoint(potentialEndpointNode)
 
+	hasHetznerNodes := k.hasHetznerNodes(data.Nodepools)
+	httpProxyMode := utils.GetEnvDefault("HTTP_PROXY_MODE", defaulHttpProxyMode)
+
+	data.NoProxy = ""
+	if httpProxyMode == "on" || (httpProxyMode != "off" && hasHetznerNodes) {
+		// Claudie utilizes proxy, because the proxy mode is either turned on,
+		// or it isn't turned off and there is at least 1 Hetzner node in the k8s cluster
+		data.UtilizeHttpProxy = true
+		// add nodes privat and public IPs to the NoProxy. Otherwise the kubeone proxy won't work properly
+		for _, nodePool := range data.Nodepools {
+			for _, node := range nodePool.Nodes {
+				if data.NoProxy == "" {
+					data.NoProxy = fmt.Sprintf("%s,%s", node.Node.Private, node.Node.Public)
+				} else {
+					data.NoProxy = fmt.Sprintf("%s,%s,%s", data.NoProxy, node.Node.Private, node.Node.Public)
+				}
```

Comment:
What you can do here is to create a slice of strings and append to that (also in the case of Lbs) and later join them with a comman, i.e.

```
var noProxy []string

for _, nodePool := range data.Nodepools {
    for _, node := range nodePool.Nodes {
       noProxy = append(noProxy, node.Node.Private, node.Node.Public)
    }
}

....

data.NoProxy = fmt.Sprintf("%s,svc", strings.Join(noProxy, ","))
```